### PR TITLE
test: add tests for non-fungible assets with Zoe contracts

### LINKED ISSUE #771: [Zoe] Add tests for Zoe contracts to use non-fungible assets (setStr and set mathHelpers)
Currently all of our contracts (apart from autoswap/uniswap, because the constant product formula only makes sense for fungible tokens with natural number amounts) should work with non-fungible tokens. We need to create test cases to show how the non-fungible tokens work in these contracts with mathHelpers `setStr` and `set`.

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
Closes #771

### CODE REVIEW

**Comment by katelynsills** [comment]
Do we have any examples with `set` mathHelpers? I saw three versions with `strSet`, but it would be nice to test both in this PR.

File: packages/zoe/test/unitTests/setupMixedMints.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+const setupMixed = () => {
+  const ccBundle = produceIssuer('CryptoCats', 'strSet');
+  const moolaBundle = produceIssuer('moola');
+  const allBundles = { cc: ccBundle, moola: moolaBundle };
+  const mints = new Map();
+  const issuers = new Map();
+  const amountMaths = new Map();
+  const brands = new Map();
+
+  for (const k of Object.getOwnPropertyNames(allBundles)) {
```
Comment (katelynsills) [inline packages/zoe/test/unitTests/setupMixedMints.js:14]: I really like how you did this! It takes the best of the array version and the best of the bundle version.

File: packages/zoe/test/unitTests/setupNonFungibleMints.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import harden from '@agoric/harden';
+
+import produceIssuer from '@agoric/ertp';
+
+const setupNonFungible = () => {
+  const ccBundle = produceIssuer('CryptoCats', 'strSet');
+  const rpgBundle = produceIssuer('MMORPG Items', 'strSet');
```
Comment (katelynsills) [inline packages/zoe/test/unitTests/setupNonFungibleMints.js]: Could one of these use a `set` mathHelper?

**Review by katelynsills — CHANGES_REQUESTED** [review]
Everything looks fantastic to me! I would just want to see a few tests with the `set` Math helpers, but everything so far is great~

File: packages/zoe/test/unitTests/setupNonFungibleMints.js (hunk 2 above)
Comment (Chris-Hibbert) [reply packages/zoe/test/unitTests/setupNonFungibleMints.js]: done

**Review by katelynsills — APPROVED** [review]
Looks great, thank you!

**Pull request merged**